google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreRpcContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.InternalApi;
+import com.google.api.core.InternalExtensionOnly;
+import com.google.api.gax.rpc.ApiStreamObserver;
+import com.google.api.gax.rpc.BidiStreamingCallable;
+import com.google.api.gax.rpc.ServerStreamingCallable;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.cloud.firestore.spi.v1.FirestoreRpc;
+
+@InternalApi
+@InternalExtensionOnly
+interface FirestoreRpcContext<FS extends Firestore> {
```

Comment:
My takeaway from our discussion last week that that this was intended to be public. Let me know if I misunderstood you :)

Reply:
I think that was my original intent, but after poking at things and seeing the amount of interconnectedness between the data and the operations I think we need to scope this to the level `FirestoreImpl` currently is. If this interface were as simple as `getDatabseName()` and `getResourcePath()` I think it'd be easier for us to have it public, but there is a lot of coupling against these methods that we don't want on the public api surface.